[Bug]: External OAuth2 provider fails when the useAsUsername value is different from email

### Installation Method

None

### The Problem

I have a problem with my Docker configuration.
I configured an external OAuth2 provider using 'mail' as useAsUsername (instead of 'email'). It worked with the 0.42 image (I'm using the fat image), but in the v0.43.2 image, it failed to start with the following error:
Caused by: java.lang.IllegalArgumentException: No enum constant stirling.software.SPDF.model.UsernameAttribute.MAIL"

I tested with different useAsUsername values (uid, nameID, etc.), but it still failed. I tested too with force settings with local file (settings.yml) and environment values.

Could someone review this and apply a fix in future versions if appropriate?

Regards,
Sergio



### Version of Stirling-PDF

0.43.2

### Last Working Version of Stirling-PDF

0.42.0

### Page Where the Problem Occurred

_No response_

### Docker Configuration

```txt
version: '3.3'
services:
  stirling-pdf:
    image: stirlingtools/stirling-pdf:latest-fat
    pull_policy: always
    ports:
      - '8080:8080'
    volumes:
      # - ./StirlingPDF/trainingData:/usr/share/tessdata # Required for extra OCR languages
      - ./StirlingPDF/extraConfigs:/configs
      - ./StirlingPDF/customFiles:/customFiles/
      # - ./StirlingPDF/logs:/logs/
      # - ./StirlingPDF/pipeline:/pipeline/
    environment:
      - DOCKER_ENABLE_SECURITY=true
```

### Relevant Log Output

```txt
Caused by: java.lang.IllegalArgumentException: No enum constant stirling.software.SPDF.model.UsernameAttribute.UID
        at java.base/java.lang.Enum.valueOf(Enum.java:293)
        at stirling.software.SPDF.model.UsernameAttribute.valueOf(UsernameAttribute.java:5)
        at stirling.software.SPDF.config.security.oauth2.OAuth2Configuration.oidcClientRegistration(OAuth2Configuration.java:188)
        at stirling.software.SPDF.config.security.oauth2.OAuth2Configuration.clientRegistrationRepository(OAuth2Configuration.java:61)
        at stirling.software.SPDF.config.security.oauth2.OAuth2Configuration$$SpringCGLIB$$0.CGLIB$clientRegistrationRepository$1(<generated>)
        at stirling.software.SPDF.config.security.oauth2.OAuth2Configuration$$SpringCGLIB$$FastClass$$1.invoke(<generated>)
        at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:258)
        at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:372)
        at stirling.software.SPDF.config.security.oauth2.OAuth2Configuration$$SpringCGLIB$$0.clientRegistrationRepository(<generated>)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at org.springframework.beans.factory.support.SimpleInstantiationStrategy.lambda$instantiate$0(SimpleInstantiationStrategy.java:171)
        ... 127 common frames omitted
```

### Additional Information

This is my settings.yml configuration (only oauth2 external provider):
`security:
  enableLogin: true # set to 'true' to enable login
  csrfDisabled: false # set to 'true' to disable CSRF protection (not recommended for production)
  loginAttemptCount: 5 # lock user account after 5 tries; when using e.g. Fail2Ban you can deactivate the function with -1
  loginResetTimeMinutes: 120 # lock account for 2 hours after x attempts
  loginMethod: oauth2 # Accepts values like 'all' and 'normal'(only Login with Username/Password), 'oauth2'(only Login with OAuth2) or 'saml2'(only Login with SAML2)
  initialLogin:
    username: '' # initial username for the first login
    password: '' # initial password for the first login
  oauth2:
    enabled: true # set to 'true' to enable login (Note: enableLogin must also be 'true' for this to work)
    client:
      keycloak:
        issuer: '' # URL of the Keycloak realm's OpenID Connect Discovery endpoint
        clientId: '' # client ID for Keycloak OAuth2
        clientSecret: '' # client secret for Keycloak OAuth2
        scopes: openid, profile, email # scopes for Keycloak OAuth2
        useAsUsername: preferred_username # field to use as the username for Keycloak OAuth2. Available options are: [email | name | given_name | family_name | preferred_name]
      google:
        clientId: '' # client ID for Google OAuth2
        clientSecret: '' # client secret for Google OAuth2
        scopes: https://www.googleapis.com/auth/userinfo.email, https://www.googleapis.com/auth/userinfo.profile # scopes for Google OAuth2
        useAsUsername: email # field to use as the username for Google OAuth2. Available options are: [email | name | given_name | family_name]
      github:
        clientId: '' # client ID for GitHub OAuth2
        clientSecret: '' # client secret for GitHub OAuth2
        scopes: read:user # scope for GitHub OAuth2
        useAsUsername: login # field to use as the username for GitHub OAuth2. Available options are: [email | login | name]
    issuer: https://OMITED.OMITED.es # set to any Provider that supports OpenID Connect Discovery (/.well-known/openid-configuration) endpoint
    clientId: pdf-OMITED # client ID from your Provider
    clientSecret: OMITED # client secret from your Provider
    autoCreateUser: true # set to 'true' to allow auto-creation of non-existing users
    blockRegistration: false # set to 'true' to deny login with SSO without prior registration by an admin
    useAsUsername: mail # default is 'email'; custom fields can be used as the username
    scopes: openid # specify the scopes for which the application will request permissions
    provider: SSO OMITED # set this to your OAuth Provider's name, e.g., 'google' or 'keycloak'`


### Browsers Affected

_No response_

### No Duplicate of the Issue

- [x] I have verified that there are no existing issues raised related to my problem.

--- a/src/main/java/stirling/software/SPDF/model/UsernameAttribute.java
+++ b/src/main/java/stirling/software/SPDF/model/UsernameAttribute.java
@@ -4,10 +4,12 @@ import lombok.Getter;
 
 @Getter
 public enum UsernameAttribute {
+    MAIL("mail"),
     EMAIL("email"),
     LOGIN("login"),
     PROFILE("profile"),
     NAME("name"),
+    UID("uid"),
     USERNAME("username"),
     NICKNAME("nickname"),
     GIVEN_NAME("given_name"),

